atomicapp/providers/docker.py

```diff
@@ -50,16 +59,74 @@ def init(self):
                    "on host (%s). Please update your host." % (client, server))
             raise ProviderFailedException(msg)
 
+    """
+    Deploys a Docker container based upon the /artifacts file located in the working dir.
+    If --name is provided within /artifacts/docker, atomicapp will tag the corresponsind image as: atomic_NAME_HASH
+    If --name is not provided within /artifacts/docker, atomicapp will use the namespace providers in answers.conf
+    If neither are provided, atomicapp will use the default name for deployment (default)
+    """
     def deploy(self):
+        logger.info("Deploying to provider: Docker")
+
         for artifact in self.artifacts:
             artifact_path = os.path.join(self.path, artifact)
             label_run = None
             with open(artifact_path, "r") as fp:
                 label_run = fp.read().strip()
+            labels = label_run.split()
+
+            # if --name is located in the artifact file, append it with the docker deployment naming convention
+            if '--name' in labels:
+                cname = labels.index('--name') + 1
+                logger.info("Using name located within artifacts/docker: %s", cname)
+                labels[cname] = "%s_%s_%s" % (self.default_name, labels[cname], Utils.getUniqueUUID())
 
-            cmd = label_run.split()
+            # otherwise, name it using the namespace located in answers.conf
+            else:
+                labels.insert(labels.index('run') + 1, "--name=%s_%s_%s" % (self.default_name, self.namespace, Utils.getUniqueUUID()))
+            cmd = labels
 
             if self.dryrun:
                 logger.info("DRY-RUN: %s", " ".join(cmd))
             else:
                 subprocess.check_call(cmd)
+
+    """
+    Undeploys a Docker container from the local machine based on the name(s) located within the workin dir answers.conf and artifacts
+    """
+    def undeploy(self):
+        logger.info("Undeploying to provider: Docker")
+        artifact_names = list()
+
+        # Gather the list of containers within /artifacts/docker
+        for artifact in self.artifacts:
+            artifact_path = os.path.join(self.path, artifact)
+            label_run = None
+            with open(artifact_path, "r") as fp:
+                label_run = fp.read().strip()
+            labels = label_run.split()
+
+            # If any artifacts are labelled by name, add it to a container dict list
+            if '--name' in labels:
+                artifact_names.append(labels[labels.index('--name') + 1])
+                logger.debug("artifact cnames: %s" % artifact_names)
+
+        # TODO: If anyone know's a cleaner way to grep all Docker containers by name; please submit a PR
```

Comment:
RTFM for docker-ps:-)

```
docker ps -a -q --filter=name=$NAME_STUB
```

where `$NAME_STUB` can be full name or just a part of it:)